### PR TITLE
Update spec tests to v0.12.3

### PR DESCRIPTION
**Pull request by terencechain**
Update spec tests to v0.12.3

Part of #7274

### CODE REVIEW

File: beacon-chain/core/epoch/precompute/spectest/rewards_penalties_test.go [hunk 1]
```diff
@@ -26,14 +26,16 @@ var deltaFiles = []string{"source_deltas.yaml", "target_deltas.yaml", "head_delt
...
 	require.NoError(t, spectest.SetConfig(t, config))
-	testPath := "rewards/core/pyspec_tests"
-	testFolders, testsFolderPath := testutil.TestFolders(t, config, testPath)
-	for _, folder := range testFolders {
-		helpers.ClearCache()
-		t.Run(folder.Name(), func(t *testing.T) {
-			folderPath := path.Join(testsFolderPath, folder.Name())
-			runPrecomputeRewardsAndPenaltiesTest(t, folderPath)
-		})
+	testPaths := []string{"rewards/basic/pyspec_tests", "rewards/leak/pyspec_tests", "rewards/random/pyspec_tests"}
```
Comment (terencechain) [inline beacon-chain/core/epoch/precompute/spectest/rewards_penalties_test.go:29]: Updated rewards paths. See: https://github.com/ethereum/eth2.0-specs/pull/1957

File: proto/testing/BUILD.bazel [hunk 2]
```diff
@@ -102,7 +102,6 @@ go_test(
     name = "go_mainnet_test",
     size = "small",
     srcs = [
-        "ssz_static_mainnet_test.go",
```
Comment (terencechain) [inline proto/testing/BUILD.bazel:105]: ssz_static tests for mainnet was deprecated. (minimal is still there). See: https://github.com/ethereum/eth2.0-specs/pull/1957

Comment (prestonvanloon) [reply proto/testing/BUILD.bazel:105]: Can you elaborate? I still see "mainnet" for phase0.yaml in that PR

Comment (prestonvanloon) [reply proto/testing/BUILD.bazel:105]: Asked upstream here: https://github.com/ethereum/eth2.0-specs/pull/1957#issuecomment-695330482

Comment (terencechain) [reply proto/testing/BUILD.bazel:105]: thanks, marking this as blocked until this is resovled

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**